ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala

```diff
@@ -273,65 +273,217 @@ final class SqlLedgerSpec
+      val timeProvider = TimeProvider.Constant(nowInstant)
 
-  private def createSqlLedger(validatePartyAllocation: Boolean): Future[Ledger] =
-    createSqlLedger(None, None, List.empty, validatePartyAllocation)
+      val minSkew = Duration.ofSeconds(10)
+      val maxSkew = Duration.ofSeconds(30)
 
-  private def createSqlLedger(
-      ledgerId: String,
-      validatePartyAllocation: Boolean,
-  ): Future[Ledger] =
-    createSqlLedger(ledgerId, List.empty, validatePartyAllocation)
+      val seedService = SeedService.WeakRandom
+      val applicationId = Ref.ApplicationId.assertFromString(UUID.randomUUID().toString)
+      val party = Ref.Party.assertFromString("party1")
+      val commandId = Ref.CommandId.assertFromString("commandId1")
+      val submissionId = Ref.SubmissionId.assertFromString("submissionId1")
+      val transactionLedgerEffectiveTime = now.add(Duration.ofMinutes(5))
+      for {
+        sqlLedger <- createSqlLedger(
+          ledgerId = None,
+          participantId = None,
+          packages = List.empty,
+          timeProvider = timeProvider,
+        )
+        start = sqlLedger.ledgerEnd()
+        _ <- sqlLedger.publishConfiguration(
+          maxRecordTime = now.add(Duration.ofMinutes(1)),
+          submissionId = "configuration",
+          config = Configuration.reasonableInitialConfiguration.copy(
+            timeModel = LedgerTimeModel.reasonableDefault.copy(minSkew = minSkew, maxSkew = maxSkew)
+          ),
+        )
+        result <- sqlLedger.publishTransaction(
+          submitterInfo = SubmitterInfo(
+            actAs = List(party),
+            applicationId = applicationId,
+            commandId = commandId,
+            deduplicationPeriod = DeduplicationPeriod.DeduplicationDuration(Duration.ofHours(1)),
+            submissionId = submissionId,
+            ledgerConfiguration = Configuration.reasonableInitialConfiguration,
+          ),
+          transactionMeta =
+            emptyTransactionMeta(seedService, ledgerEffectiveTime = transactionLedgerEffectiveTime),
+          transaction = EmptySubmitted,
+        )
+        completion <- sqlLedger
+          .completions(
+            startExclusive = Some(start),
+            endInclusive = None,
+            applicationId = domain.ApplicationId(applicationId),
+            parties = Set(party),
+          )
+          .runWith(Sink.head)
+      } yield {
+        result should be(SubmissionResult.Acknowledged)
+        completion._1 should be > start
+        inside(completion._2.completions) {
+          case Seq(Completion(`commandId`, Some(status), _, _, _, _, _)) =>
+            status.code should be(Status.Code.ABORTED.value)
+            val lowerBound = nowInstant.minus(minSkew)
+            val upperBound = nowInstant.plus(maxSkew)
+            status.message should be(
+              s"Ledger time 2021-09-01T18:05:00Z outside of range [2021-09-01T17:59:50Z, 2021-09-01T18:00:30Z]"
```

Comment:
Did you want to interpolate the actual lower and upper bound values here?

Reply:
I did, and then decided that would not be testing anything useful, so I inlined them.